### ISSUE TITLE
Need new keyboard shortcut for switching

### ISSUE THREAD

**Issue by dnschneid** [dnschneid/crouton#1386]
We're going to need the extension to process keyboard shortcuts moving forward, but Ctrl+Alt+Shift+F1/F2 won't work.

In addition to choosing new default key combos, the Chrome UI allows for the key to be changed by the user in the settings page.  We'll need a way to pass that on to the xbindkeys daemon, and a way to detect changes to these combos (probably just on focus switch).

[chrome.commands](https://developer.chrome.com/extensions/commands) lists constraints on possible key combos, but basically, it's any of A-Z, 0-9, Comma, Period, Home, End, PageUp, PageDown, Space, Insert, Delete, Arrow keys (Up, Down, Left, Right), and the modifier Ctrl, Search, Alt, Ctrl+Search or Alt+Search.  I think shift can be included as well.

Any suggestions for the new keyboard shortcut(s)?  Do we need separate front/back cycle keys, or is "cycle" sufficient?

**Comment by tedm** [comment]
why not retain xiwi's alt - tab, so cycling windows is seamless with the host windows?

**Comment by dnschneid** [comment]
Alt-tab works when xiwi is not maximized, but you generally want that passed to and handled by chroot, not used to switch environments.

**Comment by divx118** [comment]
Keeping forward/back would be nice. Maybe just a simple search combination like "search + a" back "search + s" forward.

**Comment by nahberry** [comment]
ctrl + alt + shift  + "up", ctrl + alt + shift + "down". keeping seperate front/back cycle keys is a good idea. Use standard keys available on external keyboards as well as the chromebook layout for people may be using a chromebox/external keyboard.

**Comment by nahberry** [comment]
@divx118 nope you read it right, I had it wrong. ctrl + alt is not permitted, using shift in place of alt is permitted though. That might be an option. Macbooks would automatically convert ctrl to command, but that shouldn't be an issue.

**Comment by dnschneid** [comment]
Maybe `Ctrl+Shift+[` and `Ctrl+Shift+]`?  Ctrl by itself is pretty crowded (and if we want to keep the same shortcut in the chroot, ctrl by itself is a pretty bad choice), and I understand not wanting to use search.

**Comment by dnschneid** [comment]
Oh wait, `[` and `]` aren't allowed either.

**Comment by dnschneid** [comment]
`Alt+Shift` is a LOT less crowded.  `Alt+Shift+Q` and `Alt+Shift+W`, or `A+S+9` and `A+S+0` (i.e., parentheses on the US keyboard), or `Alt+Shift+C` for crouton/chroot/Chromium....clearly I should have chosen a name starting with a different letter of the alphabet.

**Comment by DennisLfromGA** [comment]
Or `Alt+Shift+,` and `Alt+Shift+.` similar to switching between multiple profiles in ChromeOS (which is `Ctrl+Alt+,` and `Ctrl+Alt+.`)

**Comment by drinkcat** [comment]
Do we know how this plays with different keyboard layouts? That would clearly eliminate letters and symbols (comma, period), as the position of those vary widely.

I think numbers 0-9 are mostly consistent (possible exception: I believe French keyboard has numbers on Shift positions, but maybe it would work anyway as we have the shift key on).

Alt-Shift-Arrows in also tempting, but it actually does something useful in text editors...

So, bottom line, Alt-Shift-9/0 is looking good.

**Comment by drinkcat** [comment]
Whoops, will this work at all?

https://developer.chrome.com/extensions/commands:
> NOTE: The exception here is Chrome OS, where global commands are not allowed at the moment.

**Comment by dnschneid** [comment]
Oops, ok.  Returning to the back channels, I guess...
